### ISSUE TITLE
Intellisense not interpreting C++17 auto correctly

### ISSUE THREAD

**Issue by jeffbi** [microsoft/vscode-cpptools#2112]
Type: LanguageService

- OS and Version: Linux Mint 64-bit
- VS Code Version: 1.24.0
- C/C++ Extension Version: 0.17.4

Intellisense interprets the construct `auto p{"Fred"}` as deduction of an initializer list. In C++17 that should not be the case.

**To Reproduce**
Use the following C++17 `main.cpp` file:
```cpp
int main()
{
    auto s = "fred";
    auto p{s};
}
```
along with the following (auto-generated) `c_cpp_properties.json` file:
```json
{
    "configurations": [
        {
            "name": "Linux",
            "browse": {
                "path": [
                    "${workspaceFolder}"
                ],
                "limitSymbolsToIncludedHeaders": true
            },
            "includePath": [
                "${workspaceFolder}"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/gcc",
            "cStandard": "c11",
            "cppStandard": "c++17",
            "intelliSenseMode": "clang-x64"
        }
    ],
    "version": 4
}
```
Red squiggles appear under `auto p`. Intellisense is seeing this line as a deduction of an initializer list.

**Expected behavior**
`auto p{s}` should be recognized as `p` being a `const char *`.

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. It appears to be a bug with our "clang-x64" mode, and doesn't repro with the msvc-x64 mode.

**Comment by sean-mcmanus** [comment]
This appears fixed in 0.17.7 -- not sure what release actually fixed it. 0.17.8-insiders has more fixes for other scenarios: https://github.com/Microsoft/vscode-cpptools/releases/tag/v0.17.8-insiders , and we made several more parser fixes that haven't been released yet (https://github.com/Microsoft/vscode-cpptools/milestone/14 ).

**Comment by sean-mcmanus** [comment]
Correction -- it's not fixed in 0.17.7, but it is in 0.17.8-insiders. Also, your compiler library needs to be a new enough version (I tested with gcc-8) or you'll need to #include <initializer_list>.

**Comment by jeffbi** [comment]
Nice. Thank you.
